accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -298,10 +304,21 @@ fields on `Object`.
 It is an error to call an expression whose type is potentially nullable and not
 `dynamic`.
 
-It is an error if a top level variable, static variable, or instance field with
-potentially non-nullable type has no initializer expression and is not
-initialized in a constructor via an initializing formal or an initializer list
-entry, unless the variable or field is marked with the `late` modifier.
+It is an error if a top level variable or static variable with a non-nullable type 
+has no initializer expression unless the variable is marked with the `late` modifier.
+
+It is an error if a class declaration declares an instance variable with a potentially 
+non-nullable type and no initializer expression, and the class has a generative constructor 
+where the variable is not initialized via an initializing formal or an initializer list entry, 
+unless the variable is marked with the `late` modifier.
```

Comment:
It is a tricky corner case that a class `C` with no generative constructors does not allow for creation of instances of `C`, nor for declaration of subclasses of `C`, and it escapes the above rule. (It can have useful static methods, and it can be a non-superclass superinterface, and it _will_ have a factory constructor which must return one of those subtypes, so maybe someone may want to write such a thing.)

You could say that it is then no problem to have an instance variable with a potentially non-nullable type and no initializer, because there cannot ever exist an instance of `C` or any subclass thereof, so it's no problem that it has a "bad" instance variable.

Nevertheless, I'd prefer that we make this an error, because it is a confusing setup, and not a useful one (as far as I can see):

```suggestion
unless the variable is marked with the `late` modifier.
It is an error if a class declaration declares an instance variable with a potentially 
non-nullable type and no initializer expression, and the class has no generative constructors.
```

If anyone really wants that API they can declare an abstract getter and setter (or one of those newfangled abstract fields ;-) and make the class abstract, and they can then preserve the rest of the setup without changes.

Reply:
Made the class-with-no-generative-constructor and non-late-potentially-nullable-uninitialized-instance-variable combination an error too.